File: ROS2_DEVELOPMENTS/ros3_workspace4_fastapi_bridge_projects/src/robot_bridge_py/robot_bridge_py/talker.py
import rclpy
from rclpy.node import Node

# Import all constants (including MESSAGE_TYPE)
from robot_bridge_py.constants import *

class MinimalPublisher(Node):
    def __init__(self):
        # Use constant node name
        super().__init__(PUBLISHER_NODE_NAME)
        
        # Use constant message type, topic name, and queue size
        self.publisher_ = self.create_publisher(MESSAGE_TYPE, CHATTER_TOPIC, QUEUE_SIZE)
        self.timer = self.create_timer(TIMER_PERIOD, self.timer_callback)
        self.i = 0

    def timer_callback(self):
        # Create message instance of the imported MESSAGE_TYPE class
        msg = MESSAGE_TYPE()
        msg.data = PAYLOAD_FORMAT.format(self.i)
        
        self.publisher_.publish(msg)
        self.get_logger().info(LOG_PUBLISH_FORMAT.format(msg.data))
        self.i += 1

def main(args=None):
    rclpy.init(args=args)
    minimal_publisher = MinimalPublisher()
    rclpy.spin(minimal_publisher)
    minimal_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ROS2_DEVELOPMENTS/ros3_workspace4_fastapi_bridge_projects/src/robot_bridge_cpp/src/talker.cpp
#include <chrono>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

// Include the constants file
#include "constants.hpp"

using namespace std::chrono_literals;

class MinimalPublisher : public rclcpp::Node {
public:
  MinimalPublisher()
  : Node(robot_bridge_cpp::constants::PUBLISHER_NODE_NAME), count_(0) {
    
    // Create publisher using constants
    publisher_ = this->create_publisher<std_msgs::msg::String>(
      robot_bridge_cpp::constants::CHATTER_TOPIC,
      robot_bridge_cpp::constants::QUEUE_SIZE
    );
    
    // Create wall timer using constants
    timer_ = this->create_wall_timer(
      std::chrono::duration<double>(robot_bridge_cpp::constants::TIMER_PERIOD_SEC),
      std::bind(&MinimalPublisher::timer_callback, this)
    );
    
    RCLCPP_INFO(this->get_logger(), "🤖 C++ Simulated Robot Publisher Node Started");
  }

private:
  void timer_callback() {
    auto message = std_msgs::msg::String();
    message.data = "Hello World from C++: " + std::to_string(count_++);
    
    RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
    publisher_->publish(message);
  }

  rclcpp::TimerBase::SharedPtr timer_;
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  size_t count_;
};

int main(int argc, char * argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MinimalPublisher>());
  rclcpp::shutdown();
  return 0;
}
